src/db.c

```diff
@@ -3653,6 +3624,40 @@ db_pl_delete_bypath(const char *path)
   free(qp.filter);
 }
 
+int
+db_pl_purge_byid(int id)
+{
+#define Q_TMPL "DELETE FROM files WHERE path in (SELECT filepath FROM playlistitems WHERE playlistid = %d)"
+
+  char *query;
+  int ret;
+
+  query = sqlite3_mprintf(Q_TMPL, id);
+  if (!query)
+    {
+      DPRINTF(E_LOG, L_DB, "Out of memory for query string\n");
+      return -1;
+    }
+
+  db_transaction_begin();
+  ret = db_query_run(query, 1, LISTENER_DATABASE);
```

Comment:
Typically we'd let the `purge_cruft` take care of cleaning out the entries in `files` that would be part of any other `.*scan()`.  However if a user calls the library to `item_remove()` we do not have this.

For a playlist, we should remove the `files` at this time - if we dont' the tracks will still appear on the playlists screen and stay there until the next `scan`/`purge_cruft` call which is undesirable.  Therefore I am remove the `files` entries at the same time explicitly instead of using `purge_cruft()`.